src/lambda.rs

```diff
@@ -207,18 +215,17 @@ where
                 }
             };
 
-            // Process each of the records. If any of them fail, return immediately.
-            // Do this concurrently if required to saturate the CPU, control number
-            // of records by setting the event source mapping BatchSize (Default is 10).
+            // Process the records in the event batch concurrently, up to `RECORD_CONCURRENCY`.
+            // If any of them fail, return immediately.
             stream::iter(event.records)
                 .map(|record| {
                     let body = record
                         .body
                         .as_ref()
                         .with_context(|| format!("No SqsMessage body: {:?}", record))?;
-                    serde_json::from_str::<Msg>(body)
-                        .with_context(|| format!("Error parsing body into message: {}", body))
-                        .map(|msg| (msg, body.to_owned())) //Body must be passed for error message
+                    let msg = serde_json::from_str::<Msg>(body)
+                        .with_context(|| format!("Error parsing body into message: {}", body))?;
+                    Ok((msg, body.to_owned()))
```

Comment:
I find this easier to read, as it now says "try to create `body`, try to create `msg`, if both those operations succeed, pass `(msg, body)` to the next stage".